shared/actions/chat/index.js

```diff
@@ -318,7 +318,7 @@ function * _loadMoreMessages (action: Constants.LoadMoreMessages): SagaGenerator
     let newMessages = []
     messages.forEach((message, idx) => {
       if (idx > 0) {
-        const timestamp = Shared.maybeAddTimestamp(messages[idx], messages, idx - 1)
+        const timestamp = Shared.maybeAddTimestamp(messages[idx], List(messages).take(idx))
```

Comment:
i know we've had some back and forth here but this feels dirty to keep on building a longer and longer copy of the list on each iteration. Let's just do the simple for loop here and not do this